components/contribution-flow/StepPayment.js

```diff
@@ -77,7 +78,17 @@ const getPaymentMethodMetadata = pm => {
           }}
         />
       );
-    } else if (pm.expiryDate) {
+    } else if (pm.expiryDate && paymentMethodExpired(pm.expiryDate)) {
+      return (
+        <FormattedMessage
+          id="ContributePayment.expiredOn"
+          defaultMessage="Expired on {expiryDate}"
+          values={{
+            expiryDate: <FormattedDate value={pm.expiryDate} day="numeric" month="long" year="numeric" />,
+          }}
+        />
+      );
+    } else if (pm.expiryDate && !paymentMethodExpired(pm.expiryDate)) {
```

Comment:
This is not necessary, because if the payment method is expired the `else if` above will catch it

```suggestion
    } else if (pm.expiryDate) {
```